src/transport_ofi.c

```diff
@@ -591,9 +591,16 @@ static inline int atomic_limitations_check(void)
 
 static inline int exchange_and_av_insert(struct fabric_info *info)
 {
-    int                 i, rank = 0, ret = 0;
-    char   epname[128], *alladdrs = NULL;
-    size_t epnamelen = sizeof(epname);
+    int    i, rank = 0, ret = 0;
+    size_t epnamelen = 128, full_ep_len = epnamelen + HOST_NAME_MAX, adrlen = 0;
+    char   epname[full_ep_len], *alladdrs = NULL, *epnamebuff = NULL;
+#ifdef USE_ON_NODE_COMMS
+    int num_on_node = 0
+    size_t pidlen = HOST_NAME_MAX;
+    full_ep_len = pidlen;
```

Comment:
yeah I agree, I'll fix this so that the initial epname size is saved in a different variable